miio/roidmivacuum_miot.py

```diff
@@ -0,0 +1,590 @@
+        """Start cleaning."""
+        return self.call_action_by(2, 1)
+
+    @command(click.argument("roomstr", type=str))
+    def start_room_sweep_unknown(self, roomstr: str) -> None:
+        """Start cleaning.
+
+        FIXME: the syntax of voice is unknown
+        """
+        return self.call_action_by(2, 3, roomstr)
+
+    @command(
+        click.argument("sweep_mode", type=EnumType(SweepMode)),
+        click.argument("clean_info", type=str),
+    )
+    def start_sweep_unknown(self, sweep_mode: SweepMode, clean_info: str) -> None:
+        """Start sweep with mode.
+
+        FIXME: the syntax of start sweep with mode is unknown
+        """
+        return self.call_action_by(14, 1, [sweep_mode.value, clean_info])
+
+    @command()
+    def stop(self) -> None:
+        """Stop cleaning."""
+        return self.call_action_by(2, 2)
+
+    @command()
+    def home(self) -> None:
+        """Return to home."""
+        return self.call_action_by(3, 1)
+
+    @command()
+    def identify(self) -> None:
+        """Locate the device (i am here)."""
+        return self.call_action_by(8, 1)
+
+    @command(click.argument("vol", type=int))
+    def set_sound_volume(self, vol: int):
+        """Set sound volume [0-100]."""
+        return self.set_property("volume", vol)
+
+    @command(click.argument("cleaning_mode", type=EnumType(CleaningMode)))
+    def set_cleaning_mode(self, cleaning_mode: CleaningMode):
+        """Set cleaning_mode."""
+        return self.set_property("cleaning_mode", cleaning_mode.value)
+
+    @command(click.argument("sweep_type", type=EnumType(SweepType)))
+    def set_sweep_type(self, sweep_type: SweepType):
+        """Set sweep_type."""
+        return self.set_property("sweep_type", sweep_type.value)
+
+    @command(click.argument("path_mode", type=EnumType(PathMode)))
+    def set_path_mode(self, path_mode: PathMode):
+        """Set path_mode."""
+        return self.set_property("path_mode", path_mode.value)
+
+    @command(click.argument("work_station_freq", type=int))
+    def set_work_station_freq(self, work_station_freq: int):
+        """Set work_station_freq (2 means Auto dust colect every second time)."""
+        return self.set_property("work_station_freq", work_station_freq)
+
+    @command(click.argument("timing", type=str))
+    def set_timing_unknown(self, timing: str):
+        """Set time zone.
+
+        FIXME: the syntax of timing is unknown
+        """
+        return self.set_property("timing", timing)
+
+    @command(click.argument("auto_boost", type=bool))
+    def set_auto_boost(self, auto_boost: bool):
+        """Set auto boost on carpet."""
+        return self.set_property("auto_boost", auto_boost)
+
+    @command(
+        click.argument("begin", type=str),
+        click.argument("end", type=str),
+        click.argument("active", type=bool, required=False, default=True),
+    )
+    def set_forbid_mode(self, begin: str, end: str, active: bool = True):
```

Comment:
```suggestion
    def set_dnd(self, start_hr: int, start_min: int, end_hr: int, end_min: int, active: bool = True)
```
This would make it similar to vacuum.py's `set_dnd` with the added `active` flag, for future proofing :-)